### DISCUSSION TITLE
WPF Application takes too much RAM usage than Winform, how can I optimize WPF App's memory usage?

### DISCUSSION THREAD

**Discussion by neville-studio** [dotnet/csharplang#8049] [General] [answered]
I have used .NET 8.0 to build two blank applications, the one is WPF application, the another one is WinForm, then publish these applications.

The WPF Application takes too much memory, more than 90MB memory used, but the WinForm application only takes 6.0MB - 7.0 MB memory usage.

So I want to ask how to optimize WPF Application's memory usage if I want to develop a tool.

**Comment by HaloFour** [comment, marked as answer]
This repository is specifically about the design and evolution of the C# language, not any particular runtime or UX framework.  You might have better luck with their respective repositories:  [WinForms](https://github.com/dotnet/winforms) and [WPF](https://github.com/dotnet/wpf)

I'd suggest that the biggest difference is likely that WinForms is a relatively thin wrapper over the Win32APIs, which are the native UX framework in Windows and have been around in one form or another for almost the past 40 years.  WPF is built on DirectX and designed to be much, much more flexible.

https://www.geeksforgeeks.org/difference-between-wpf-and-winforms/#

**Reply by neville-studio** [reply]
Thanks.
